### ISSUE TITLE
Sometimes duplicated popup for authorizing domains from iframe in Firefox

### ISSUE THREAD

**Issue by pabzm** [mailvelope/mailvelope#841]
I'm seeing two popup windows for authorizing a domain from an injected iframe in Firefox 126.0.1 on macos 14.5.

It doesn't work on all sites, I can't tell the cause for that.

To reproduce go with Firefox to e.g. `https://mailvelope.github.io/mailvelope/`, open the Web Developer Tools, paste this into the console and press Enter:

```javascript
const iframe = document.createElement('iframe'); iframe.style = 'display: none'; iframe.src = "https://api.mailvelope.com/authorize-domain/?api=true"; document.body.append(iframe)
```

In my case the expected popup is opened, but twice.

Interestingly this doesn't happen on `https://mailvelope.github.io/`, and neither on `https://demo.mailvelope.com/`. On many other sites it does happen.

On all sites I tested this doesn't happen with Google Chrome (v126.0.6478.62), Microsoft Edge (v126.0.2592.56), or Vivaldi (Chromium-based, v6.7.3329.41), so I assume it is related to Firefox.

I have no Linux- or MS Windows-device to test on.

**Comment by pabzm** [comment]
One more detail: if a Content-Security-Policy is in place, that prevents the iframe's src to be attempted to be loaded (e.g. `frame-src 'self';`, then only one popup window is opened.

Some sites, where the described reproducing process does not work, appear to not have a CSP in place, so I guess this info is only a partial reason.

**Comment by toberndo** [comment]
I was able to reproduce the duplicate popup windows. But I was not able to find out exactly why this is happening or how to prevent it with different methods to load the iframe. Some [reports](https://bugzilla.mozilla.org/show_bug.cgi?id=583351) indicate that a duplicate loading of resources cannot be ruled out. So we need to fix this on the extension side and I will implement a check that prevents opening multiple of these popups.

**Comment by toberndo** [comment]
This is fixed now with release v5.2.0
